object/slicer/slicer_test.go

```diff
@@ -854,3 +855,101 @@ func TestSlicedObjectsHaveSplitID(t *testing.T) {
 		}
 	})
 }
+
+func BenchmarkWritePayloadBuffer(b *testing.B) {
+	for _, tc := range []struct {
+		sizeLimit uint64
+		size      uint64
+	}{
+		{sizeLimit: 1 << 10, size: 1},
+		{sizeLimit: 1 << 10, size: 1 << 10},
+		{sizeLimit: 1 << 10, size: 10 << 10},
+		{sizeLimit: 1 << 10, size: 200 << 10},
+		{sizeLimit: 1 << 26, size: 1 << 10},
+	} {
+		b.Run(fmt.Sprintf("limit=%d,size=%d", tc.sizeLimit, tc.size), func(b *testing.B) {
+			ctx := context.Background()
+			in, opts := randomInput(b, tc.size, tc.sizeLimit)
+			obj := objecttest.Object(b)
+			hdr := *obj.CutPayload()
+
+			b.Run("with payload buffer", func(b *testing.B) {
+				opts := opts
+				opts.SetPayloadBuffer(make([]byte, tc.sizeLimit+1))
+
+				b.ReportAllocs()
```

Comment:
Can we _never_ have this line in our tests? It's just a `-benchmem` flag to me.

Reply:
yes we can, create an issue pls. To me, tests in which the main indicator is memory should be equipped with this method. It makes no sense to run such tests without memory measurements